### PR TITLE
Fix bug in JetPlusTrackCorrections_cff.py: add JetVertexExtender

### PR DESCRIPTION
**Pull request by kodolova**
JetPlusTrack correction practically does not work and return calojet energy only without adding ak4JetExtender to the python configurator.

### CODE REVIEW

File: RecoJets/JetPlusTracks/python/JetPlusTrackCorrections_cff.py [hunk 1]
```diff
@@ -41,7 +42,10 @@
 
 JetPlusTrackCorrectionsAntiKt4 = cms.Sequence(
     JPTeidTight*
+    #trackExtrapolator*
```
Comment (slava77) [inline RecoJets/JetPlusTracks/python/JetPlusTrackCorrections_cff.py]: please remove the commented out line, unless there's a good reason for it to be here

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
Hi Olga,

I see that ak4JetExtenderJPT is added (it produces a JetExtendedAssociation::Container), but nothing is reading its data in the jobs we are running and it's data is not saved.
Either this PR is not complete (a keep statement is needed) or I'm missing something.

Please clarify.

**Comment by slava77** [comment]
Just to clarify the status from my side: I'm not planning to sign off on addition of a module which doesn't make any used products.
So, I'm waiting for feedback.

**Comment by kodolova** [comment]
Hi Slava, you are right. Only ak4JetTracksAssociatorAtCaloFace is needed to be added.
I removed JetExtender and checked that JPT correction is ok. sorry for inconvenience.

**Comment by slava77** [comment]
Hi Olga,
ak4JetTracksAssociatorAtCaloFace is already defined and is running prior to JetPlusTrackZSPCorJetAntiKt4 as a part of 
ak4JTA https://github.com/cms-sw/cmssw/blob/CMSSW_7_6_X/RecoJets/JetAssociationProducers/python/ak4JTA_cff.py

In which workflow did you see an issue?
Is it in the full RECO sequence or just some smaller JPT analysis job?

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@cmsbuild please test
the comparison job failed

@smuzaffar  please check that we don't have some general infrastructure issues (a noticeable fraction of comparison jobs show up in red in jenkins)

**Comment by slava77** [comment]
@cmsbuild please test

... the last one failed with 

```
ssl.SSLError: _ssl.c:493: The handshake operation timed out
Build step 'Execute shell' marked build as failure
```

**Comment by kodolova** [comment]
Hi Slava, I found the issue when I run JPT reconstruction over AODSIM trying to derive L2L3 corrections.

**Comment by slava77** [comment]
On 8/31/15 12:12 PM, kodolova wrote:
> Hi Slava, I found the issue when I run JPT reconstruction over AODSIM
> trying to derive L2L3 corrections.

Olga, thank you for the clarification.
It makes sense to be an issue in a setup like that
and your fix is appropriate.

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/10977	 7d9a2b5
- the change is good for applications starting with JetPlusTrackCorrectionsAntiKt4 sequence (full reco sequence is not affected because the missing module is present in ak4JTA included elsewhere
- jenkins tests pass and comparisons with baseline show no relevant differences
    - DQM plots show differences in PU ID variables, which seem random and show up in  CMSSW_7_6_X_2015-08-30-2300 in unrelated places as  #11053 . I'm running valgrind to find the origin.
- tested locally in CMSSW_7_6_X_2015-08-30-2300 /test area sign581/: the same modules are executed in the same order in this PR and in the baseline

**Comment by davidlange6** [comment]
+1

**Pull request merged**